staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/statusreporting/BatchStatusProvider.java

```diff
@@ -0,0 +1,135 @@
+                final StagingException stagingException
+                    = new StagingException("Failed to get progress report from directory: " + inProgressBatchDir, ex);
+                progressReportExceptions.add(stagingException);
+            }
+        }
+
+        // Add in the progress reports from this service
+        inProgressReports.addAll(BatchProgressTracker.get(tenantId, batchId));
+
+        // Check if the batch has been completed
+        final boolean batchInCompletedState = batchPathProvider.getPathForBatch(tenantId, batchId).toFile().exists();
+
+        if (!batchInCompletedState) {
+            // Ignore IOExceptions that happened when retrieving the in-progress reports if the batch is completed,
+            // as they were probably due to the batch completing while the report was being collected.
+            if (!progressReportExceptions.isEmpty()) {
+                for (final StagingException ex : progressReportExceptions) {
+                    LOGGER.error("Failed to get progress report", ex);
+                }
+                throw progressReportExceptions.get(0);
+            }
+
+            if (inProgressReports.isEmpty()) {
+                throw new BatchNotFoundException(batchId.getValue());
+            }
+        }
+
+        assert batchInCompletedState || !inProgressReports.isEmpty();
+        assert batchInCompletedState || progressReportExceptions.isEmpty();
+
+        // Build and return the Response
+        final InProgress inProgress = new InProgress();
+        inProgress.setMetrics(inProgressReports.stream().map(Tracker::toInProgressMetrics).collect(Collectors.toList()));
+        final BatchStatus batchStatus = new BatchStatus();
+        batchStatus.setBatchComplete(batchInCompletedState);
+        batchStatus.setInProgress(inProgress);
+        final BatchStatusResponse response = new BatchStatusResponse();
+        response.setBatchId(batchId.getValue());
+        response.setBatchStatus(batchStatus);
+        return response;
+    }
+
+    private List<Path> getOtherServicesInProgressBatchDirectories(final TenantId tenantId, final BatchId batchId) throws StagingException
+    {
+        final Path tenantInProgressDir = batchPathProvider.getTenantInprogressDirectory(tenantId);
+        final String batchIdStr = batchId.getValue();
+        final List<Path> list = new ArrayList<>();
+        try (final DirectoryStream<Path> directoryStream = Files.newDirectoryStream(tenantInProgressDir)) {
+            for (final Path batch : directoryStream) {
+                final BatchInProgressDirectoryName batchDir = new BatchInProgressDirectoryName(batch.getFileName().toString());
+                if (!batchDir.isThisService() && batchDir.getBatchId().equals(batchIdStr)) {
+                    list.add(batch);
+                }
+            }
+        } catch (final IOException e) {
+            LOGGER.error("Error while traversing In-progress folder for the tenant: " + tenantId, e);
```

Comment:
This line should have been removed when the exception was added, because the exception is caught and logged.  Leaving this log statement will cause double-logging of the error.

(The logging in the `getStatus()` method above is because there may be multiple exceptions in the collection and we are only going to throw one of them.  There is double-logging of the first one but it is not worth the extra code to prevent it.)

Reply:
**Resolved in this commit:**  
[b60a18bec0e53e0b468bc8604ad947afbd22107e] Drop the double-logging